ppmpy/ppm.py

```diff
@@ -904,22 +904,33 @@ def compute_N2(self, fname, num_type='ndump', radeos=True):
 
         return N2
 
-    def compute_m(self, fname, num_type='ndump'):
+    def compute_m(self, fname, num_type='ndump', minner=0):
         if self.__isyprofile:
             r = self.get00('Y', fname, num_type=num_type, resolution='l')
             rho = self.get('Rho', fname, num_type=num_type, resolution='l')
+            rinner = 0
+            minner = 0
 
         if self.__isRprofSet:
             r = self.get('R', fname, num_type=num_type, resolution='l')
             rho = self.get('Rho0', fname, num_type=num_type, resolution='l') + \
                   self.get('Rho1', fname, num_type=num_type, resolution='l')
-
-        dm = -4.*np.pi*r**2*cdiff(r)*rho
-        m = np.cumsum(dm)
-
-        # We store everything from the surface to the core.
-        m = m[-1] - m
-        self.__messenger.warning('WARNING: PPMtools.compute_m() integrates mass from r = 0. This will not work for shell setups and wrong gravity will be returned.')
+            rinner = self.get('radin0', fname, num_type=num_type, resolution='l')
+        
+        if minner==0 and rinner==0: # core setup
+            dm = -4.*np.pi*r**2*cdiff(r)*rho
+            m = np.cumsum(dm)
+            # We store everything from the surface to the core.
+            m = m[-1] - m 
+        elif minner>0 and rinner>0: # shell setup
+            rho_int = rho[::-1]
+            r_int = r[::-1]
+            rho_int[r_int<rinner] = 0
+            dm = 4.*np.pi*r_int**2*cdiff(r_int)*rho_int
+            m = np.cumsum(dm) + minner
+            m = m[::-1]
+        else:
+            raise IOError("Shell setup detected, please specify minner; see m0 in setup file")
```

Comment:
pls use `class Messenger` of `ppm.py` for issuing error messages, look around for examples

also send "message" from `class Messenger` when using minner so that we are reminded that there should be a change in ppm code to make the use of this parameter unnecessary